dartsim/src/EntityManagementFeatures.cc

```diff
@@ -665,6 +658,50 @@ bool EntityManagementFeatures::ModelRemoved(const Identity &_modelID) const
   return !this->models.HasEntity(_modelID);
 }
 
+/////////////////////////////////////////////////
+bool EntityManagementFeatures::RemoveNestedModelByIndex(
+    const Identity &_modelID, std::size_t _nestedModelIndex)
+{
+  auto modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  if (_nestedModelIndex >= modelInfo->nestedModels.size())
+  {
+    return this->GenerateInvalidId();
+  }
+  const auto nestedModelID = modelInfo->nestedModels[_nestedModelIndex];
+  if (this->models.HasEntity(nestedModelID))
+  {
+    const auto worldID = this->GetWorldOfModelImpl(nestedModelID);
+    const auto model = this->models.at(nestedModelID)->model;
+    const auto filterPtr = GetFilterPtr(this, worldID);
+    filterPtr->RemoveSkeletonCollisions(model);
+    return this->RemoveModelImpl(worldID, nestedModelID);
+  }
+  return false;
+}
+
+/////////////////////////////////////////////////
+bool EntityManagementFeatures::RemoveNestedModelByName(const Identity &_modelID,
+                                                 const std::string &_modelName)
+{
+  auto modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  const std::string fullName =
+      ::sdf::JoinName(modelInfo->model->getName(), _modelName);
+
+  if (this->models.HasEntity(_modelID))
+  {
+    auto worldID = this->GetWorldOfModelImpl(_modelID);
+    auto nestedSkel = this->worlds.at(worldID)->getSkeleton(fullName);
+    if (nullptr == nestedSkel || !this->models.HasEntity(nestedSkel))
+    {
+      return false;
+    }
+    const std::size_t nestedModelID = this->models.IdentityOf(nestedSkel);
+    const auto filterPtr = GetFilterPtr(this, worldID);
+    filterPtr->RemoveSkeletonCollisions(nestedSkel);
+    return this->RemoveModelImpl(worldID, nestedModelID);
```

Comment:
The logic here is pretty similar to `RemoveNestedModelByIndex`, I think this function could call that one.

Reply:
I don't think it would be trivial to be able to call `RemoveNestedModelByIndex` because we have to find the index based on the name first. Ideally  the lines 
```
    const auto filterPtr = GetFilterPtr(this, worldID);
    filterPtr->RemoveSkeletonCollisions(nestedSkel);
```
could go into `RemoveModelImpl`, but that would involve moving `GetFilterPtr` into `Base.hh`. How about we revisit this later? I can leave a TODO for myself in the code.

Reply:
Yeah later is fine, it was just a small suggestion